pkg/controller/operands/kubevirtConfigMap_test.go

```diff
@@ -0,0 +1,212 @@
+package operands
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/pkg/apis/hco/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/controller/common"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/controller/commonTestUtils"
+	hcoutils "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+var _ = Describe("Test kubeVirtCmHandler", func() {
+	var (
+		hco            *hcov1beta1.HyperConverged
+		req            *common.HcoRequest
+		expectedEvents = []commonTestUtils.MockEvent{
+			{
+				EventType: corev1.EventTypeNormal,
+				Reason:    "Killing",
+				Msg:       fmt.Sprintf("Removed ConfigMap %s", kvCmName),
+			},
+		}
+		emitter = commonTestUtils.NewEventEmitterMock()
+	)
+
+	BeforeEach(func() {
+		hco = commonTestUtils.NewHco()
+		req = commonTestUtils.NewReq(hco)
+
+		req.SetUpgradeMode(false)
+
+		emitter.Reset()
+	})
+
+	It("should do nothing if the configMap does not exist", func() {
+		c := commonTestUtils.InitClient([]runtime.Object{hco})
+		handler := newKubeVirtCmHandler(c, emitter)
+
+		res := handler.ensure(req)
+
+		Expect(res.Err).ToNot(HaveOccurred())
+		Expect(res.UpgradeDone).To(BeTrue())
+
+		Expect(emitter.CheckEvents(expectedEvents)).To(BeFalse())
+	})
+
+	It("should remove the configMap if it exists", func() {
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      kvCmName,
+				Namespace: commonTestUtils.Namespace,
+			},
+			Data: map[string]string{
+				"fakeKey": "fakeValue",
+			},
+		}
```

Comment:
The block of the config-map definition is duplicated 4 times in this file, for almost every test case (4 out of 5).
Why not defining it in the top, under `var` of the `Describe` block?

Reply:
Common var is not an option because some cases are desructive (e.g. the client.Get function).

I can use a NewKvConfig function instead.